app/factories/shopify/variant_factory.rb

```diff
@@ -0,0 +1,31 @@
+module Shopify
+  class VariantFactory
+    def initialize(spree_variant, shopify_variant)
+      @spree_variant = spree_variant
+      @shopify_variant = shopify_variant
+    end
+
+    def perform
+      shopify_variant.weight = spree_variant.weight
+      shopify_variant.weight_unit = 'oz'
+      shopify_variant.price = spree_variant.price
+      shopify_variant.sku = spree_variant.sku
+      shopify_variant.updated_at = spree_variant.updated_at
+      generate_options!
+
+      shopify_variant
+    end
+
+    private
+
+    attr_accessor :spree_variant, :shopify_variant
+
+    def generate_options!
+      shopify_variant.option1 = spree_variant.sku
+
+      spree_variant.option_values.each_with_index do |option, index|
+        shopify_variant.send("option#{index + 2}=", option.name)
```

Comment:
I gotta tell you, this response blew my mind a bit.

Why would you double the maintenance workload by writing comments instead of self-documenting code? This introduces a risk of desynchronizing the code and the documentation. An additional benefit of writing expressive code is its expressiveness exists in two places – where the methods exist, and where the methods are invoked. The code can become almost human-readable, which is so much friendlier for future maintainers and integrators.